### PR TITLE
fix: [CI-5251]: Create triggers one after the other instead of at the same time

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
### Summary

This change fixes https://harness.atlassian.net/browse/CI-5251.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11176)

**Review by d1wilko — CHANGES_REQUESTED** [review]
Let me know your thoughts :)

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx [hunk 1]
```diff
@@ -291,28 +291,31 @@ export const InfraProvisioningWizard: React.FC<InfraProvisioningWizardProps> = p
...
-                          projectIdentifier,
-                          pipelineIdentifier: createPipelineResponse?.data?.identifier,
-                          stageId: getString('buildText')
-                        })
-                      )
-                    }
+              createPRTrigger
+                .then((createPRTriggerResponse: ResponseNGTriggerResponse) => {
+                  if (createPRTriggerResponse?.status === Status.SUCCESS) {
+                    createPushTrigger.then((createPushTriggerResponse: ResponseNGTriggerResponse) => {
```
[the remark this answers is not in the document]
Comment (vardanbansal-harness) [reply src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx]: Done.

**Review by d1wilko — APPROVED** [review]
LGTM :)

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx [hunk 2]
```diff
@@ -291,13 +291,14 @@ export const InfraProvisioningWizard: React.FC<InfraProvisioningWizardProps> = p
...
-                    createPushTriggerResponse?.status === Status.SUCCESS &&
-                    createPRTriggerResponse?.status === Status.SUCCESS
-                  ) {
+              createPRTrigger
+                .then((createPRTriggerResponse: ResponseNGTriggerResponse) => {
+                  if (createPRTriggerResponse?.status === Status.SUCCESS) {
+                    return createPushTrigger
+                  }
+                })
+                .then((createPushTriggerResponse?: ResponseNGTriggerResponse) => {
```
Comment (mtran7) [inline src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx:300]: are we handling error case gracefully?

**Review by mtran7 — APPROVED** [review]

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx (hunk 2 above)
Comment (vardanbansal-harness) [reply src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx:300]: Yeah if either of these promises fail, the chained `.catch` should be able to handle it.

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (vardanbansal-harness) [reply src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx]: Thanks for reminding me of chaining though, I forgot it existed 😅

**Pull request merged**